### PR TITLE
Reject the `index_options` parameter for numeric fields

### LINKED ISSUE #21475: Numeric fields should not accept the `index_options` parameter
Numeric fields no longer support `index_options`, so this parameter should be deprecated and removed.

### PR DESCRIPTION
**Pull request by cbuescher**
Numeric fields no longer support the `index_options` parameter. This changes the parameter 
to be rejected in numeric field types after it was deprecated in 6.0.

Closes #21475

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java]: I will open a separate PR with deprecation logging instead of the error against the 6.x branches.

Comment (cbuescher) [reply core/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java]: I opened https://github.com/elastic/elasticsearch/pull/26672 for the deprecation.

File: core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java
Comment (jpountz) [inline core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java]: Could we override the `indexOptions` method in the number field type instead of introducing a new method?

**Review by jpountz — APPROVED** [review]
I left a suggestion.

File: core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java
Comment (cbuescher) [reply core/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java]: Sure, I changed this. Will wait for another CI run before I merge.

**Pull request merged**
